src/utils/cacheClientUtils.ts

```diff
@@ -112,3 +112,5 @@ export const createInMemoryCacheClientWithNamespace = <T>(namespace: string) =>
     );
 
 export const getResultsFromCache = async (key: string): Promise<CachedResult> => getCacheStrategyInstance().getValueFromCache(CacheNamespaces.RESULTS_NAMESPACE, key);
+
+export const refreshTtlForCachedResult = async <T>(key: string, ttl: number, value: CachedResult<T>): Promise<CachedResult> => getCacheStrategyInstance().refreshTtlForCachedResult(CacheNamespaces.RESULTS_NAMESPACE, key, ttl, value);
```

Comment:
Please fix here return type into Promise<void> (same as it is abstract class)

Reply:
I also wonder about making this function not work async. Why? Refreshing ttl is not that important for performance. It might be also done after the parent function (for example `execQueryWithCache`) return the value. 
Imagine a situation where there is 50 reads of a given cache key within 5 seconds.  That would be quite unnecessary to call `refreshTtlForCachedResult()` 49 extra times - as we only need one in that case. 


I have two good options here - and mix of both will have the best performance here. 

a). Make this function non-async and wrap it with some `setTimeout(() => {} ,0)` to push it further in the event loop.

b). Create a queueWorker for this task, which will make sure that we will only execute it once. There is  https://github.com/arqo123/speedgoose/blob/master/src/utils/queueUtils.ts
which  already contain something similar. It can work like 

1. Call `refreshTtlForCachedResult()`
2. Inside `refreshTtlForCachedResult()` make logic to append the task of clearing cache into the worker queue.
3. In the worker queue, check if the task already exists in queue. If not - then add it to the queue. 

If something is not clear just let me know.

Reply:
cool! I'll try with this implementation